### ISSUE TITLE
(uf-ho) Fatal failure at src/theory/theory_model.cpp:690 

### ISSUE THREAD

**Issue by zhendongsu** [cvc5/cvc5#6536]
Commit: https://github.com/cvc5/cvc5/commit/31242de4b423d7225174dd1672edb2dacb68f5b8
OS: Ubuntu 18.04
Possibly related: https://github.com/cvc5/cvc5/issues/5371

```
[775] % cvc4 --strings-exp --uf-ho -q small.smt2
Fatal failure within void cvc5::theory::TheoryModel::assignFunctionDefinition(cvc5::Node, cvc5::Node) at /local/suz-local/software/CVC4/src/theory/theory_model.cpp:690
Check failure

 f_def.isConst()
Non-constant f_def: (lambda (($x1 b) ($x2 b)) (let ((_let_1 (h (h e)))) (ite (and (= _let_1 $x1) (= _let_1 $x2)) (h _let_1) e)))
Aborted
[776] % 
[776] % cat small.smt2
(declare-datatypes ((a 0) (b 0)) (((c) (d)) ((h (j b)) (e))))
(declare-fun f () b)
(declare-fun k (a) b)
(declare-fun g (b b) b)
(assert (forall ((i a)) (distinct (k i) (ite ((_ is c) i) e (ite ((_ is d) i) (h (g (k  i) (k  i))) f)))))
(check-sat)
[777] % 
```

**Comment by HanielB** [comment]
This issue no longer occurs. Probably fixed by #6697, which improved the interplay between the HOL setup and the rest of the solver (particularly with `--strings-exp`).

**Comment by HanielB** [comment]
The failure is still triggered in debug.
